app/components/viral/modal_component.html.erb

```diff
@@ -0,0 +1,71 @@
+<div data-controller="viral--modal-component">
+  <!-- Main modal -->
+  <div
+    data-viral--modal-component-target="modal"
+    tabindex="-1"
+    aria-hidden="true"
+    aria-label="Modal"
```

Comment:
**chore (blocking):** Sorry, I thought I wrote this here, but it has disappeared so if it repeats twice sorry.  By setting `aria-label` to be "Modal" it describes all modals that way instead of something explicit like "Confirm deletion of project?".  A better way to handle this can be found by looking at [MDN's `aria-modal`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal) write up and example:

```html
<div id="backdrop" class="no-scroll">
  <div
    role="alertdialog"
    aria-modal="true"
    aria-labelledby="dialog_label"
    aria-describedby="dialog_desc">
    <h2 id="dialog_label">Confirmation</h2>
    <div id="dialog_desc">
      <p>Are you sure you want to delete this file?</p>
    </div>
    <button type="button" onclick="closeDialog(this)">
      No. Close this popup.
    </button>
    <button type="button" onclick="deleteFile(this)">
      Yes. Delete the file.
    </button>
  </div>
</div>
```

- `role="alertdialog"` : From [MDN ARIA: alertdialog role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alertdialog_role) - "The alertdialog role is to be used on modal alert dialogs that interrupt a user's workflow to communicate an important message and require a response."
- `aria-labelledby`: They use the id for the modal title, which is probably a great way to set it.